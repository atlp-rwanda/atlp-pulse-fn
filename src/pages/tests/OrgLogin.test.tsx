import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client';
import { fireEvent, render } from '@testing-library/react';
import OrgLogin from '../Organization/Orglogin';

const client = new ApolloClient({ cache: new InMemoryCache() });

describe('Organization Login', () => {
  it('Should render', () => {
    const elem = renderer
      .create(
        <ApolloProvider client={client}>
          <MemoryRouter>
            <OrgLogin />
          </MemoryRouter>
        </ApolloProvider>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});

it('Should submit the form', async () => {
  const { getByTestId, findByText } = render(
    <ApolloProvider client={client}>
      <MemoryRouter>
        <OrgLogin />
      </MemoryRouter>
    </ApolloProvider>,
  );

  fireEvent.input(getByTestId('orgName'), {
    target: { value: 'Andela' },
  });
  fireEvent.submit(getByTestId('loginForm'));
});
