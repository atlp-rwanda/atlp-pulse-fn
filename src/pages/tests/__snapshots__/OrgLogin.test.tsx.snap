// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization Login Should render 1`] = `
<div
  className="grow  dark:bg-dark-frame-bg bg-indigo-50 flex flex-row font-sans"
>
  <div
    className="hidden lg:flex w-[50%] h-full  flex-col justify-center items-center mt-10 "
  >
    <div
      className=" flex flex-col justify-center items-center "
    >
      <div
        className="relative flex flex-row justify-center"
      >
        <img
          alt="pulses"
          className="absolute inset-0 w-[41rem] h-[12rem] dark:hidden"
          src={Object {}}
        />
        <p
          className="relative w-[70%] text-gray-700 text-[1.9em] p-5 text-center italic font-bold  dark:text-dark-text-fill"
        >
          Boost your organization
        </p>
      </div>
      <div
        className="w-[30vw] h-[42vh]  flex flex-row "
      >
        <img
          alt="functions"
          className="w-sm dark:brightness-150"
          src={Object {}}
        />
      </div>
    </div>
  </div>
  <div
    className=" w-full  lg:w-[50%] lg:p-5 flex flex-row items-end p-5"
  >
    <div
      className="max-w-lg w-full mx-auto my-28 bg-indigo-100 dark:bg-dark-bg  p-5 sm:p-14 md:shadow-xl sm:shadow-none md:rounded-xl sm:rounded-none"
    >
      <div
        className="text-center  text-black-600 text-2xl font-bold dark:text-dark-text-fill "
      >
        Sign in to your Organization
      </div>
      <div
        className="text-md  text-black-600 mt-2 text-center font-semibold dark:text-dark-text-fill sm:text-xs"
      >
        Enter your organizationâ€™s Dev-Pulse URL
      </div>
      <form
        action="#none"
        className="space-y-6 mt-4"
        onSubmit={[Function]}
      >
        <label
          className="orgName bg-white dark:bg-neutral-600 rounded mt-1 py-1 px-2"
          htmlFor=":r0:"
        >
          <input
            className="inputStyle inputOne w-full border-primary rounded dark:bg-neutral-600"
            data-testid="orgName"
            id=":r0:"
            name="name"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Your-organization"
            style={
              Object {
                "width": "130px",
              }
            }
            type="text"
            value=""
          />
          <span
            className="inputStyle inputTwo w-full p-2 border border-primary rounded text-gray-400"
          >
            .devpulse.org
          </span>
        </label>
        <div>
          <button
            className="btn primary lg w-full ml-0 hover:bg-violet-600 text-sm"
            data-testid="loginForm"
            disabled={false}
            type="submit"
          >
             
            Continue
          </button>
        </div>
        <div
          className="w-full text-sm  text-light-text dark:text-dark-text-fill"
        >
          Looking to register an organization instead?
          <a
            className="text-primary"
            href="/signup/org"
            onClick={[Function]}
          >
            Register a new organization
          </a>
        </div>
      </form>
    </div>
  </div>
</div>
`;
